xdem/coreg.py

```diff
@@ -1453,24 +1452,27 @@ def apply_matrix(dem: np.ndarray, transform: rio.transform.Affine, matrix: np.nd
     # Create a skimage-compatible array of the new index coordinates that the pixels shall have after warping.
     inds = np.vstack((y_inds.reshape((1,) + y_inds.shape), x_inds.reshape((1,) + x_inds.shape)))
 
-    # Warp the DEM
-    transformed_dem = skimage.transform.warp(
-        filled_dem,
-        inds,
-        order=resampling_order,
-        mode="constant",
-        cval=0,
-        preserve_range=True
-    )
-    # Warp the NaN mask, setting true to all values outside the new frame.
-    tr_nan_mask = skimage.transform.warp(
-        nan_mask.astype("uint8"),
-        inds,
-        order=resampling_order,
-        mode="constant",
-        cval=1,
-        preserve_range=True
-    ) > 0.25  # Due to different interpolation approaches, everything above 0.25 is assumed to be 1 (True)
+    with warnings.catch_warnings():
+        # An skimage warning that will hopefully be fixed soon. (2021-07-30)
+        warnings.filterwarnings("ignore", message="Passing `np.nan` to mean no clipping in np.clip")
+        # Warp the DEM
+        transformed_dem = skimage.transform.warp(
+            filled_dem,
+            inds,
+            order=resampling_order,
+            mode="constant",
+            cval=np.nan,
+            preserve_range=True
+        )
+        # Warp the NaN mask, setting true to all values outside the new frame.
+        tr_nan_mask = skimage.transform.warp(
+            nan_mask.astype("uint8"),
+            inds,
+            order=resampling_order,
+            mode="constant",
+            cval=1,
+            preserve_range=True
+        ) > 0.1  # Due to different interpolation approaches, everything above 0.25 is assumed to be 1 (True)
```

Comment:
You've changed the threshold value but not the comment.

Reply:
Fixed in 468029f